include/parakeet/UdpSocket.h

```diff
@@ -0,0 +1,72 @@
+/*
+	Copyright 2021 OpenJAUS, LLC (dba MechaSpin). Subject to the MIT license.
+*/
+
+#ifndef PARAKEET_ETHERNETPORT_H
+#define PARAKEET_ETHERNETPORT_H
+
+#include <string> 
+#include <cstring>
+#include <thread>
+#include <chrono>
+
+namespace mechaspin
+{
+namespace parakeet
+{
+class UdpSocket
+{
+public:
+	UdpSocket() = default;
+
+	/// \brief Open a UDP Socket for reading
+	/// \param[in] ipAddress - The IP Address of the device to communicate with
+	/// \param[in] srcPort - The port which this device will read messages from
+	/// \returns If opening the port was successful
+	bool open(const char* ipAddress, int srcPort);
+
+	/// \brief Close the existing UDP Socket
+	void close();
+
+	/// \brief Read data from an opened UDP socket
+	/// \param[in] buffer - The buffer in-which read data will be placed
+	/// \param[in] currentLength - The number of bytes used in the current buffer
+	/// \param[in] bufferSize - The maximum size of the buffer
+	/// \returns The number of bytes read from the stream
+	int read(unsigned char* buffer, int currentLength, int bufferSize);
```

Comment:
For read we have an unsigned char* and for write we have a (signed) char*, is there a reason for that? I can't see where it would make a difference as at this point I just want to send the bytes on the wire, I'm not interpreting it.